File: Vanilla/sources/zdoc/media/lua/shared/Library/WornItem.lua
---@class WornItem : zombie.characters.WornItems.WornItem
---@field protected location String
---@field protected item InventoryItem
WornItem = {}

---@public
---@return InventoryItem
function WornItem:getItem() end

---@public
---@return String
function WornItem:getLocation() end

File: Vanilla/sources/zomboid/zombie/characters/WornItems/WornItem.java
package zombie.characters.WornItems;

import zombie.inventory.InventoryItem;

public final class WornItem {
   protected final String location;
   protected final InventoryItem item;

   public WornItem(String var1, InventoryItem var2) {
      if (var1 == null) {
         throw new NullPointerException("location is null");
      } else if (var1.isEmpty()) {
         throw new IllegalArgumentException("location is empty");
      } else if (var2 == null) {
         throw new NullPointerException("item is null");
      } else {
         this.location = var1;
         this.item = var2;
      }
   }

   public String getLocation() {
      return this.location;
   }

   public InventoryItem getItem() {
      return this.item;
   }
}
